### PR TITLE
Add an nginx reverse proxy to the docker container for enabling Brotli compression of outgoing responses 

### PR DESCRIPTION
**Pull request by atarashansky**
#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

**Comment by atarashansky** [comment]
Working example: https://nginx-br-rdev-explorer.rdev.single-cell.czi.technology/d/964e6017-0c94-4948-846f-33dd3c67a7fb.cxg/

File: hosted/Dockerfile [hunk 1]
```diff
@@ -32,6 +31,43 @@ WORKDIR server/eb
 RUN make build-ecs
 WORKDIR /
 
+#nginx config
+ENV NGINX_VERSION=1.21.6
+RUN wget  http://nginx.org/download/nginx-$NGINX_VERSION.tar.gz && \
```
Comment (atolopko-czi) [inline hosted/Dockerfile:36]: not: waaaay too many spaces here. 😆

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 2]
```diff
@@ -32,7 +32,7 @@ locals {
     "https://", module.explorer_dns[0].dns_prefix, ".", local.external_dns
   ]), var.frontend_url)
   explorer_image_repo = local.secret["ecrs"]["explorer"]["url"]
-  explorer_cmd        = ["gunicorn", "--worker-class", "gevent", "--bind", "0.0.0.0:5000", "server.eb.app:application", "--timeout", "60"]
+  explorer_cmd        = ["/start.sh"]
```
Comment (atolopko-czi) [inline .happy/terraform/modules/ecs-stack/main.tf:35]: ❤️

File: hosted/Dockerfile [hunk 3]
```diff
@@ -32,6 +31,43 @@ WORKDIR server/eb
 RUN make build-ecs
 WORKDIR /
 
+#nginx config
```
Comment (atolopko-czi) [inline hosted/Dockerfile:34]: add a comment noting why we must build from src, and that is ultimately all for the sake of supporting compression

File: hosted/Dockerfile [hunk 4]
```diff
@@ -32,6 +31,43 @@ WORKDIR server/eb
...
+    make modules
+
+RUN mkdir /etc/nginx/modules && \
+    mkdir /etc/nginx/sites-available && \
+    mkdir /etc/nginx/sites-enabled && \
+    cd nginx-$NGINX_VERSION && \
+    cp objs/ngx_http_brotli_filter_module.so /etc/nginx/modules/ngx_http_brotli_filter_module.so && \
+    cp objs/ngx_http_brotli_static_module.so /etc/nginx/modules/ngx_http_brotli_static_module.so && \
+    chmod 644 /etc/nginx/modules/ngx_http_brotli_filter_module.so && \
+    chmod 644 /etc/nginx/modules/ngx_http_brotli_static_module.so
```
Comment (atolopko-czi) [inline hosted/Dockerfile:65]: is it reasonable to place all of the nginx build cmds into a script and just call it? if only to reduce the visual complexity of the dockerfile.

**Review by atolopko-czi — COMMENTED** [review]
a few minor suggestions.  and does it work in rdev?

**Pull request closed without merging**